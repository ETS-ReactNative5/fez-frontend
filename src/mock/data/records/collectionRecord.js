import { hydrateMock } from '../../hydrateMock';

const collectionRecord = {
    rek_pid: 'UQ:11398',
    rek_title: 'This is a Collection Record',
    rek_display_type: 9,
    rek_display_type_lookup: 'Collection',
    rek_status: 2,
    rek_status_lookup: 'Published',
    rek_object_type: 2,
    rek_object_type_lookup: 'Collection',
    fez_record_search_key_xsd_display_option: [263],
    rek_wok_doc_type_lookup: 'Abstract of Published Item',
};
export default hydrateMock(collectionRecord);
