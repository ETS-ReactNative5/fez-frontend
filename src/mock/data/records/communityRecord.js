import { hydrateMock } from '../../hydrateMock';

const communityRecord = {
    "rek_pid": "UQ:3883",
    "rek_title": "The University of Queensland Library",
    "rek_description": null,
    "rek_display_type": 11,
    "rek_status": 2,
    "rek_date": null,
    "rek_object_type": 1,
    "rek_security_inherited": 0,
    "rek_security_policy": 5,
    "rek_display_type_lookup": "Community",
    "rek_object_type_lookup": "Community",
    "rek_status_lookup": "Published",
};
export default hydrateMock(communityRecord).data;
