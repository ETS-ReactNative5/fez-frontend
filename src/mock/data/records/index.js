export { default as publicationTypeListAudio } from './publicationTypeListAudio';
export { default as publicationTypeListBook } from './publicationTypeListBook';
export { default as publicationTypeListBookChapter } from './publicationTypeListBookChapter';
export { default as publicationTypeListConferencePaper } from './publicationTypeListConferencePaper';
export { default as publicationTypeListConferenceProceedings } from './publicationTypeListConferenceProceedings';
export { default as publicationTypeListCreativeWork } from './publicationTypeListCreativeWork';
export { default as publicationTypeListDepartmentTechnicalReport } from './publicationTypeListDepartmentTechnicalReport';
export { default as publicationTypeListImage } from './publicationTypeListImage';
export { default as publicationTypeListJournal } from './publicationTypeListJournal';
export { default as publicationTypeListManuscript } from './publicationTypeListManuscript';
export { default as publicationTypeListNewspaperArticle } from './publicationTypeListNewspaperArticle';
export { default as publicationTypeListPatent } from './publicationTypeListPatent';
export { default as publicationTypeListPreprint } from './publicationTypeListPreprint';
export { default as publicationTypeListResearchReport } from './publicationTypeListResearchReport';
export { default as publicationTypeListSeminarPaper } from './publicationTypeListSeminarPaper';
export { default as publicationTypeListThesis } from './publicationTypeListThesis';
export { default as publicationTypeListVideo } from './publicationTypeListVideo';
export { default as publicationTypeListWorkingPaper } from './publicationTypeListWorkingPaper';
export { default as collectionRecord } from './collectionRecord';
export { default as collections } from './collections';
export { default as collectionsByCommunity } from './collectionsByCommunity';
export { default as communityRecord } from './communityRecord';
export { default as communitySearchList } from './communitySearchList';
export { default as externalDoiSearchResultList } from './externalDoiSearchResultList';
export { default as externalPubMedSearchResultsList } from './externalPubMedSearchResultsList';
export { default as externalTitleScopusResultsList } from './externalTitleScopusResultsList';
export { default as externalTitleSearchResultsList } from './externalTitleSearchResultsList';
export { default as incompleteNTROlist } from './incompleteNTROlist';
export { default as incompleteNTROrecord } from './incompleteNTROrecord';
export { default as incompleteNTROrecordUqrdav10 } from './incompleteNTRORecordUqrdav10';
export { default as incompleteNTROrecordUqsbutl1 } from './incompleteNTRORecordUqsbutl1';
export { default as internalTitleSearchList } from './internalTitleSearchList';
export { default as internalTitleSearchListNoResults } from './internalTitleSearchListNoResults';
export { default as lookupToolIncites } from './lookupToolIncites';
export { default as MyDatasetList } from './MyDatasetList';
export { default as myRecordsList } from './myRecordsList';
export { default as possibleUnclaimedList } from './possibleUnclaimedList';
export { default as record } from './record';
export { default as recordsTypeList } from './recordsTypeList';
export { default as recordWithDatastreams } from './recordWithDatastreams';
export { default as recordWithoutAuthorIds } from './recordWithoutAuthorIds';
export { default as recordWithTiffAndThumbnail } from './recordWithTiffAndThumbnail';
export { default as UQ352045 } from './incompleteNTRORecordUQ352045';
export { default as UQ716942uqagrinb } from './incompleteNTRORecordUQ716942_uqagrinb';
export { default as UQ716942uqagrinbGrants } from './incompleteNTRORecordUQ716942_uqagrinb_grants';
