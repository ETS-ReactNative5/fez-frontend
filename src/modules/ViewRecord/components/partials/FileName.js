import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import ExternalLink from 'modules/SharedComponents/ExternalLink/components/ExternalLink';
import AudioPlayer from './AudioPlayer';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import Typography from '@material-ui/core/Typography';

export default class FileName extends PureComponent {
    static propTypes = {
        pid: PropTypes.string.isRequired,
        fileName: PropTypes.string.isRequired,
        mimeType: PropTypes.string.isRequired,
        mediaUrl: PropTypes.string.isRequired,
        previewMediaUrl: PropTypes.string.isRequired,
        onFileSelect: PropTypes.func.isRequired,
        allowDownload: PropTypes.bool
    };

    isAudio = (mimeType) => {
        return mimeType.indexOf('audio') >= 0;
    }

    isVideo = (mimeType) => {
        return mimeType.indexOf('video') >= 0;
    }

    isImage = (mimeType) => {
        return mimeType.indexOf('image') === 0;
    }

    canShowPreview = (mimeType) => {
        return this.isImage(mimeType);
        // TODO revert once videos are transcoded to open format #158519502
        // return (this.isImage(mimeType) || this.isVideo(mimeType));
    }

    showPreview = (mediaUrl, previewMediaUrl, mimeType) => (e) => {
        e.preventDefault();
        this.props.onFileSelect(mediaUrl, previewMediaUrl, mimeType);
    }

    render() {
        const {pid, fileName, allowDownload, mimeType, mediaUrl, previewMediaUrl} = this.props;

        return (
            <Grid container alignItems="center" wrap="nowrap">
                <Grid item>
                    {
                        allowDownload && !this.canShowPreview(mimeType) &&
                        <Typography variant="body2">
                            <ExternalLink href={mediaUrl} title={fileName} className={'fileName'} openInNewIcon>
                                {fileName}
                            </ExternalLink>
                        </Typography>
                    }
                    {
                        allowDownload && this.canShowPreview(mimeType) &&
                        <Typography variant="body2">
                            <a
                                onClick={this.showPreview(mediaUrl, previewMediaUrl, mimeType)}
                                onKeyPress={this.showPreview(mediaUrl, previewMediaUrl, mimeType)}
                                className={'fileName'}>
                                {fileName}
                            </a>
                        </Typography>
                    }
                    {
                        !allowDownload &&
                        <Typography variant="body2">{fileName}</Typography>
                    }
                </Grid>
                <Hidden xsDown>
                    <Grid item sm>
                        {
                            allowDownload && this.isAudio(this.props.mimeType) &&
                            <AudioPlayer pid={pid} fileName={fileName} mimeType={mimeType} />
                        }
                    </Grid>
                </Hidden>
            </Grid>
        );
    }
}
