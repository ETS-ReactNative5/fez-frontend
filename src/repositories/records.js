import {post, patch} from './generic';
import * as routes from './routes';

/**
 * Posts a records item
 * @param {object} data to be posted, refer to backend API
 * @returns {Promise}
 */
export function postRecord(data) {
    if (data.rek_pid) return Promise.resolve(data);
    return post(routes.NEW_RECORD_API(), data);
}

/**
 * Patches a records item
 * @param {string} pid of object to be updated
 * @param {object} data to be posted, refer to backend API
 * @returns {Promise}
 */
export function patchRecord(pid, data) {
    return patch(routes.EXISTING_RECORD_API({pid: pid}), data);
}
