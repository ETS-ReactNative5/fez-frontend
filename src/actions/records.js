import {post, patch} from 'repositories/generic';
import * as routes from 'repositories/routes';
import {putUploadFiles} from '../repositories';
import * as transformers from './transformers';
import {NEW_RECORD_DEFAULT_VALUES} from 'config/general';
import * as actions from './actionTypes';

/**
 * Save a new record involves up to three steps: create a new record, upload files, update record with uploaded files.
 * If error occurs on any stage failed action is dispatched
 * @param {object} data to be posted, refer to backend API
 * @returns {promise} - this method is used by redux form onSubmit which requires Promise resolve/reject as a return
 */
export function createNewRecord(data) {
    return dispatch => {
        dispatch({type: actions.CREATE_RECORD_SAVING});

        // set default values, links
        const recordRequest = {
            ...NEW_RECORD_DEFAULT_VALUES,
            ...JSON.parse(JSON.stringify(data)),
            ...transformers.getRecordLinkSearchKey(data),
            ...transformers.getRecordAuthorsSearchKey(data.authors || data.currentAuthor),
            ...transformers.getRecordAuthorsIdSearchKey(data.authors || data.currentAuthor),
            ...transformers.getRecordContributorsSearchKey(data.editors),
            ...transformers.getRecordContributorsIdSearchKey(data.editors),
            ...transformers.getRecordSupervisorsSearchKey(data.supervisors),
            ...transformers.getRecordSubjectSearchKey(data.fieldOfResearch)
        };

        // delete extra form values from request object
        if (recordRequest.authors) delete recordRequest.authors;
        if (recordRequest.editors) delete recordRequest.editors;
        if (recordRequest.files) delete recordRequest.files;
        if (recordRequest.currentAuthor) delete recordRequest.currentAuthor;
        if (recordRequest.supervisors) delete recordRequest.supervisors;
        if (recordRequest.fieldOfResearch) delete recordRequest.fieldOfResearch;

        let newRecord = null;
        const hasFilesToUpload = data.files && data.files.queue && data.files.queue.length > 0;
        const recordPatch = hasFilesToUpload ? {...transformers.getRecordFileAttachmentSearchKey(data.files.queue)} : null;

        return post(routes.NEW_RECORD_API(), recordRequest)
            .then(response => {
                // save new record response
                newRecord = response.data;
                return response;
            })
            .then(() =>(hasFilesToUpload ? putUploadFiles(newRecord.rek_pid, data.files.queue, dispatch) : newRecord))
            .then(() => (hasFilesToUpload ? patch(routes.EXISTING_RECORD_API({pid: newRecord.rek_pid}), recordPatch) : newRecord))
            .then((response) => {
                dispatch({
                    type: actions.CREATE_RECORD_SUCCESS,
                    payload: response.data ? response.data : newRecord
                });
                return Promise.resolve(response.data ? response.data : newRecord);
            })
            .catch(error => {
                // record was created, but file upload or record patch failed
                if (!!newRecord && !!newRecord.rek_pid) {
                    dispatch({
                        type: actions.CREATE_RECORD_SUCCESS,
                        payload: {
                            newRecord: newRecord,
                            fileUploadFailed: true
                        }
                    });

                    return Promise.resolve(newRecord);
                }

                dispatch({
                    type: actions.CREATE_RECORD_FAILED,
                    payload: error.message
                });

                return Promise.reject(error);
            });
    };
}


/**
 * Submit thesis involves two steps: upload files, create record with uploaded files.
 * If error occurs on any stage failed action is dispatched
 * @param {object} data to be posted, refer to backend API
 * @returns {promise} - this method is used by redux form onSubmit which requires Promise resolve/reject as a return
 */
export function submitThesis(data, author) {
    return dispatch => {
        const hasFilesToUpload = data.files && data.files.queue && data.files.queue.length > 0;
        if (!hasFilesToUpload) {
            // reject thesis submission, files are required
            return Promise.reject('Please attach files to proceed with thesis submission');
        }
        // set default values, links
        const recordRequest = {
            ...JSON.parse(JSON.stringify(data)),
            ...transformers.getRecordAuthorsSearchKey(data.currentAuthor),
            ...transformers.getRecordAuthorsIdSearchKey(data.currentAuthor),
            ...transformers.getRecordSupervisorsSearchKey(data.supervisors),
            ...transformers.getRecordSubjectSearchKey(data.fieldOfResearch),
            ...transformers.getRecordFileAttachmentSearchKey(data.files.queue)
        };

        // delete extra form values from request object
        if (recordRequest.authors) delete recordRequest.authors;
        if (recordRequest.editors) delete recordRequest.editors;
        if (recordRequest.files) delete recordRequest.files;
        if (recordRequest.currentAuthor) delete recordRequest.currentAuthor;
        if (recordRequest.supervisors) delete recordRequest.supervisors;
        if (recordRequest.fieldOfResearch) delete recordRequest.fieldOfResearch;

        let fileUploadSucceeded = false;
        dispatch({type: actions.CREATE_RECORD_SAVING});

        // debugger;
        return putUploadFiles(`UQ:${author.aut_student_username}`, data.files.queue, dispatch)
            .then((response) => {
                fileUploadSucceeded = !!response;
                return post(routes.NEW_RECORD_API(), recordRequest);
            })
            .then(response => {
                dispatch({
                    type: actions.CREATE_RECORD_SUCCESS,
                    payload: {
                        newRecord: response
                    }
                });
                return response;
            })
            .catch(error => {
                const specificError = !fileUploadSucceeded
                    ? 'File upload failed. Issue has been created to notify eSpace administrators. '
                    : 'Error occurred while saving record to eSpace. ';
                const compositeError = `${specificError} ${ error.message ? `(${error.message})` : '' }`;

                dispatch({
                    type: actions.CREATE_RECORD_FAILED,
                    payload: compositeError
                });

                return Promise.reject(compositeError);
            });
    };
}
/**
 * Clear new record
 * @returns {action}
 */
export function clearNewRecord() {
    return dispatch => {
        dispatch({
            type: actions.CREATE_RECORD_RESET
        });
    };
}
