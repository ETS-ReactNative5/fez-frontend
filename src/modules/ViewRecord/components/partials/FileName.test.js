import React from 'react';
import { render, fireEvent, act } from 'test-utils';

import FileName from './FileName';

import { journalArticle } from 'mock/data/testing/records';

import mediaQuery from 'css-mediaquery';
const createMatchMedia = width => {
    return query => ({
        matches: mediaQuery.match(query, { width }),
        addListener: () => {},
        removeListener: () => {},
    });
};

function setup(testProps = {}) {
    const { previewFileName, ...rest } = testProps;
    const props = {
        id: 'test-file-name',
        classes: {},
        pid: journalArticle.rek_pid,
        fileName: journalArticle.fez_record_search_key_file_attachment_name[2].rek_file_attachment_name,
        mimeType: 'application/pdf',
        mediaUrl:
            (!!testProps.fileName && `https://espace.library.uq.edu.au/view/UQ:676287/${testProps.fileName}`) || '',
        previewMediaUrl:
            (!!previewFileName && `https://espace.library.uq.edu.au/view/UQ:676287/${previewFileName}`) || '',
        onFileSelect: jest.fn(),
        allowDownload: false,
        ...rest,
    };
    return render(<FileName {...props} />);
}

describe('File Name Component ', () => {
    beforeAll(() => {
        window.matchMedia = createMatchMedia(window.innerWidth);
    });

    it('should render component and display file name only', () => {
        const { getByText, queryByTestId } = setup({});
        expect(queryByTestId('test-file-name')).toBeInTheDocument();
        expect(getByText('UQ676287_OA.pdf')).toBeInTheDocument();
    });

    it('should display file name link', () => {
        const { getByText, container } = setup({
            allowDownload: true,
            fileName: 'test.jpg',
            previewFileName: 'preview_test.jpg',
            checksums: { media: '111' },
        });
        expect(getByText('test.jpg')).toBeInTheDocument();
        expect(container.querySelector('a[title="test.jpg"][target="_blank"]')).toBeInTheDocument();
    });

    it('should render audio player', () => {
        const { getByText, queryByTestId, container } = setup({
            allowDownload: true,
            mimeType: 'audio/mp3',
            fileName: 'test.mp3',
            checksums: { media: '111' },
        });
        expect(getByText('test.mp3')).toBeInTheDocument();
        expect(queryByTestId('audioPlayer')).toBeInTheDocument();
        expect(container.querySelector('button[aria-label^="Click to play audio file"]')).toBeInTheDocument();
    });

    it('should run onFileSelect function on click', () => {
        const testFn = jest.fn();
        const { getByText } = setup({
            allowDownload: true,
            mimeType: 'image/jpeg',
            fileName: 'test.jpg',
            previewFileName: 'preview_test.jpg',
            onFileSelect: testFn,
        });
        act(() => {
            fireEvent.click(getByText('test.jpg'));
        });
        expect(testFn).toHaveBeenCalledTimes(1);
    });

    it('should run onFileSelect function on key press', () => {
        const testFn = jest.fn();
        const { getByText } = setup({
            allowDownload: true,
            fileName: 'test.mp4',
            previewFileName: 'preview_test.jpg',
            mimeType: 'video/mp4',
            onFileSelect: testFn,
        });
        act(() => {
            fireEvent.keyPress(getByText('test.mp4'), { key: 'Enter', code: 13, charCode: 13 });
        });
        expect(testFn).toHaveBeenCalledTimes(1);
    });
});
