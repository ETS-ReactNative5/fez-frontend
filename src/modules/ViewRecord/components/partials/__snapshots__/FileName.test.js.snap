// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Name Component  should display file name link 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <FileName
        allowDownload={true}
        fileName="test.jpg"
        mediaUrl="https://espace.library.uq.edu.au/view/UQ:676287/test.jpg"
        mimeType="application/pdf"
        onFileSelect={[MockFunction]}
        pid="UQ:676287"
        previewMediaUrl="https://espace.library.uq.edu.au/view/UQ:676287/preview_test.jpg"
      >
        <div
          className="columns is-gapless is-mobile fileDetails is-vcentered"
        >
          <div
            className="column fileInfo is-vcentered"
          >
            <ExternalLink
              className="fileName"
              href="https://espace.library.uq.edu.au/view/UQ:676287/test.jpg"
              openInNewIcon={true}
              rel="noopener noreferrer"
              target="_blank"
              title="test.jpg"
            >
              <a
                className="fileName externalLink showIcon"
                href="https://espace.library.uq.edu.au/view/UQ:676287/test.jpg"
                rel="noopener noreferrer"
                tabIndex="0"
                target="_blank"
                title="test.jpg"
              >
                test.jpg
              </a>
            </ExternalLink>
          </div>
        </div>
      </FileName>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`File Name Component  should render audio player 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <FileName
        allowDownload={true}
        fileName="test.mp3"
        mediaUrl="https://espace.library.uq.edu.au/view/UQ:676287/test.mp3"
        mimeType="audio/mp3"
        onFileSelect={[MockFunction]}
        pid="UQ:676287"
        previewMediaUrl=""
      >
        <div
          className="columns is-gapless is-mobile fileDetails is-vcentered"
        >
          <div
            className="column fileInfo is-vcentered"
          >
            <ExternalLink
              className="fileName"
              href="https://espace.library.uq.edu.au/view/UQ:676287/test.mp3"
              openInNewIcon={true}
              rel="noopener noreferrer"
              target="_blank"
              title="test.mp3"
            >
              <a
                className="fileName externalLink showIcon"
                href="https://espace.library.uq.edu.au/view/UQ:676287/test.mp3"
                rel="noopener noreferrer"
                tabIndex="0"
                target="_blank"
                title="test.mp3"
              >
                test.mp3
              </a>
            </ExternalLink>
          </div>
          <div
            className="column is-narrow audioWrapper is-hidden-mobile is-vcentered"
          >
            <AudioPlayer
              fileName="test.mp3"
              mimeType="audio/mp3"
              pid="UQ:676287"
            >
              <div>
                <audio>
                  <source
                    src="http://localhost/view/UQ:676287/test.mp3"
                    type="audio/mp3"
                  />
                </audio>
                <IconButton
                  aria-label="Click to play audio file test.mp3"
                  className="audioButton play"
                  disableTouchRipple={false}
                  disabled={false}
                  iconStyle={Object {}}
                  onClick={[Function]}
                  tooltipPosition="bottom-center"
                  touch={true}
                >
                  <EnhancedButton
                    aria-label="Click to play audio file test.mp3"
                    centerRipple={true}
                    className="audioButton play"
                    containerElement="button"
                    disableTouchRipple={false}
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOut={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "fontSize": 0,
                        "height": 48,
                        "overflow": "visible",
                        "padding": 12,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "width": 48,
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <button
                      aria-label="Click to play audio file test.mp3"
                      className="audioButton play"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOut={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                          "background": "none",
                          "border": 10,
                          "boxSizing": "border-box",
                          "cursor": "pointer",
                          "display": "inline-block",
                          "fontFamily": "Roboto, sans-serif",
                          "fontSize": 0,
                          "fontWeight": "inherit",
                          "height": 48,
                          "margin": 0,
                          "muiPrepared": true,
                          "outline": "none",
                          "overflow": "visible",
                          "padding": 12,
                          "position": "relative",
                          "textDecoration": "none",
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "verticalAlign": null,
                          "width": 48,
                        }
                      }
                      tabIndex={0}
                      type="button"
                    >
                      <TouchRipple
                        abortOnScroll={true}
                        centerRipple={true}
                        key="touchRipple"
                      >
                        <div
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                        >
                          <AvPlayArrow
                            key=".0"
                            style={Object {}}
                          >
                            <AvPlayArrow
                              style={Object {}}
                            >
                              <SvgIcon
                                onMouseEnter={[Function]}
                                onMouseLeave={[Function]}
                                style={Object {}}
                                viewBox="0 0 24 24"
                              >
                                <svg
                                  onMouseEnter={[Function]}
                                  onMouseLeave={[Function]}
                                  style={
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "display": "inline-block",
                                      "fill": "currentColor",
                                      "height": 24,
                                      "muiPrepared": true,
                                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                      "userSelect": "none",
                                      "width": 24,
                                    }
                                  }
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M8 5v14l11-7z"
                                  />
                                </svg>
                              </SvgIcon>
                            </AvPlayArrow>
                          </AvPlayArrow>
                        </div>
                      </TouchRipple>
                    </button>
                  </EnhancedButton>
                </IconButton>
              </div>
            </AudioPlayer>
          </div>
        </div>
      </FileName>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`File Name Component  should render component and display file name only 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <FileName
        allowDownload={false}
        fileName="UQ676287_OA.pdf"
        mediaUrl=""
        mimeType="application/pdf"
        onFileSelect={[MockFunction]}
        pid="UQ:676287"
        previewMediaUrl=""
      >
        <div
          className="columns is-gapless is-mobile fileDetails is-vcentered"
        >
          <div
            className="column fileInfo is-vcentered"
          >
            UQ676287_OA.pdf
          </div>
        </div>
      </FileName>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`File Name Component  should run onFileSelect function on click 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <FileName
        allowDownload={true}
        fileName="test.jpg"
        mediaUrl="https://espace.library.uq.edu.au/view/UQ:676287/test.jpg"
        mimeType="image/jpeg"
        onFileSelect={[MockFunction]}
        pid="UQ:676287"
        previewMediaUrl="https://espace.library.uq.edu.au/view/UQ:676287/preview_test.jpg"
      >
        <div
          className="columns is-gapless is-mobile fileDetails is-vcentered"
        >
          <div
            className="column fileInfo is-vcentered"
          >
            <a
              className="fileName"
              onClick={[Function]}
              onKeyPress={[Function]}
            >
              test.jpg
            </a>
          </div>
        </div>
      </FileName>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`File Name Component  should run onFileSelect function on key press 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <FileName
        allowDownload={true}
        fileName="test.jpg"
        mediaUrl="https://espace.library.uq.edu.au/view/UQ:676287/test.jpg"
        mimeType="image/jpeg"
        onFileSelect={[MockFunction]}
        pid="UQ:676287"
        previewMediaUrl="https://espace.library.uq.edu.au/view/UQ:676287/preview_test.jpg"
      >
        <div
          className="columns is-gapless is-mobile fileDetails is-vcentered"
        >
          <div
            className="column fileInfo is-vcentered"
          >
            <a
              className="fileName"
              onClick={[Function]}
              onKeyPress={[Function]}
            >
              test.jpg
            </a>
          </div>
        </div>
      </FileName>
    </Router>
  </MemoryRouter>
</Provider>
`;
